### DISCUSSION TITLE
[Compilers]<Cant compile sample box upgradeable contract>

### DISCUSSION THREAD

**Discussion by FurkanSezal** [zkSync-Community-Hub/zksync-developers#88] [Compilers] [answered]
### Which compiler are you using?

- [X] zksolc
- [ ] zkvyper

### Which version of Solidity or Vyper?

solidity ^0.8.16

### zksolc/zkvyper version?

latest ^0.4.1

### Explain the issue

```bash
When i try to compile sample box contract with hardhat-zksync-upgradable plug in and open zeppelin packages i get this compilier error: 
An unexpected error occurred:

Error: Cannot find module '@openzeppelin/upgrades-core/dist/ast-dereferencer'
Require stack:
- /home/furkansezal/skyRock/faucetpage/zksyncContract/node_modules/@matterlabs/hardhat-zksync-upgradable/dist/src/core/validate.js
- /home/furkansezal/skyRock/faucetpage/zksyncContract/node_modules/@matterlabs/hardhat-zksync-upgradable/dist/src/index.js
- /home/furkansezal/skyRock/faucetpage/zksyncContract/hardhat.config.js
- /home/furkansezal/skyRock/faucetpage/zksyncContract/node_modules/hardhat/internal/core/config/config-loading.js
- /home/furkansezal/skyRock/faucetpage/zksyncContract/node_modules/hardhat/internal/cli/cli.js
- /home/furkansezal/skyRock/faucetpage/zksyncContract/node_modules/hardhat/internal/cli/bootstrap.js
    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
    at Function.Module._load (node:internal/modules/cjs/loader:778:27)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.<anonymous> (/home/furkansezal/skyRock/faucetpage/zksyncContract/node_modules/@matterlabs/hardhat-zksync-upgradable/src/core/validate.ts:3:1)
    at Module._compile (node:internal/modules/cjs/loader:1103:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1157:10)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/home/furkansezal/skyRock/faucetpage/zksyncContract/node_modules/@matterlabs/hardhat-zksync-upgradable/dist/src/core/validate.js',
    '/home/furkansezal/skyRock/faucetpage/zksyncContract/node_modules/@matterlabs/hardhat-zksync-upgradable/dist/src/index.js',
    '/home/furkansezal/skyRock/faucetpage/zksyncContract/hardhat.config.js',
    '/home/furkansezal/skyRock/faucetpage/zksyncContract/node_modules/hardhat/internal/core/config/config-loading.js',
    '/home/furkansezal/skyRock/faucetpage/zksyncContract/node_modules/hardhat/internal/cli/cli.js',
    '/home/furkansezal/skyRock/faucetpage/zksyncContract/node_modules/hardhat/internal/cli/bootstrap.js'
  ]
}
```

### Link to your repo?

https://github.com/FurkanSezal/zksync-proxy

**Comment by dutterbutter** [comment]
Hello @FurkanSezal  👋,

Thank you so much for reaching out and providing a minimal example of the issue. It made it really easy to reproduce!! 🛠️

I've noticed there were updates to [astDereferencer](https://github.com/OpenZeppelin/openzeppelin-upgrades/commit/d6a23932ea74fd451d61d83521cf0d9251bcfddc) which seem to have affected our plugin. Big thanks 🙌 for highlighting this – we'll jump on it and address the issue asap. 🚀

Stay tuned for updates and thanks again for your invaluable feedback!

**Reply by FurkanSezal** [reply]
Thank you for the fast reply @dutterbutter 
What versions should i use for work it out , for now?

**Reply by kiriyaga** [reply, marked as answer]
Hello @FurkanSezal 

Check this out https://github.com/matter-labs/hardhat-zksync/issues/361

**Comment by vatosherrod** [comment]
The error message indicates that the hardhat-zksync-upgradable plugin is unable to find the '@openzeppelin/upgrades-core/dist/ast-dereferencer' module. This may be due to an issue with the plugin's dependencies or configuration. You may need to ensure that all dependencies are correctly installed and configured, and consider checking for any compatibility issues between the plugin and other packages or dependencies in your project. Additionally, you can try updating the plugin or reaching out to the plugin maintainers for further assistance.

**Comment by beckyhaynes** [comment]
It seems like there's an issue with the @matterlabs/hardhat-zksync-upgradable plugin not being able to find the @openzeppelin/upgrades-core module. This could be due to a missing or corrupted installation of the dependency.

To resolve this issue, you can try the following steps:

Reinstall Dependencies: Delete the node_modules folder and the package-lock.json or yarn.lock file, then reinstall dependencies by running npm install or yarn install.

Check npm/yarn Versions: Ensure that you're using compatible versions of npm/yarn and node.js. Sometimes, using outdated versions can lead to dependency resolution issues.

Update Plugin: Check if there's a newer version of the @matterlabs/hardhat-zksync-upgradable plugin available and update to the latest version if possible. Newer versions may have bug fixes or updated dependencies.

Verify Package Installation: After reinstalling dependencies, make sure that the @openzeppelin/upgrades-core module is properly installed in the node_modules folder. You should see it listed in the dependencies of your project.

Check Configuration: Ensure that your Hardhat configuration (hardhat.config.js) is set up correctly and that all necessary plugins are imported and configured properly.

If the issue persists after trying these steps, you may need to investigate further by checking for any compatibility issues between the plugin versions and your project dependencies. Additionally, you can reach out to the plugin maintainers or community for assistance in resolving the issue.

**Comment by hanjv** [comment]
The compilation error indicates that the module '@openzeppelin/upgrades-core/dist/ast-dereferencer' cannot be found. Ensure that you have installed the required dependencies correctly and that there are no issues with your node_modules directory. You may also want to check if there are any compatibility issues between the versions of the packages you are using. Additionally, consider reaching out to the plugin maintainers or the community for further assistance.

**Comment by jwitte3** [comment]
It seems there is an issue with the hardhat-zksync-upgradable plugin not being able to find the '@openzeppelin/upgrades-core' module during compilation. This could be due to a misconfiguration or a missing dependency in your project setup. You may need to check your dependencies and configurations to ensure everything is properly set up.
